tools/plugins/preinstall/src/main/kotlin/net/corda/cli/plugins/preinstall/CheckKafka.kt

```diff
@@ -0,0 +1,273 @@
+package net.corda.cli.plugins.preinstall
+
+import net.corda.cli.plugins.preinstall.PreInstallPlugin.PluginContext
+import net.corda.cli.plugins.preinstall.PreInstallPlugin.Kafka
+import net.corda.cli.plugins.preinstall.PreInstallPlugin.ReportEntry
+import net.corda.cli.plugins.preinstall.CheckKafka.KafkaProperties.TruststoreNotFoundException
+import org.apache.kafka.clients.admin.AdminClient
+import org.apache.kafka.common.KafkaException
+import org.apache.kafka.common.Node
+import picocli.CommandLine
+import picocli.CommandLine.Option
+import picocli.CommandLine.Parameters
+import java.util.Properties
+import java.util.concurrent.Callable
+import java.util.concurrent.ExecutionException
+
+@CommandLine.Command(name = "check-kafka", description = ["Check that Kafka is up and that the credentials work."])
+class CheckKafka : Callable<Int>, PluginContext() {
+
+    @Parameters(
+        index = "0",
+        description = ["YAML file containing the Kafka, SASL, and TLS configurations"]
+    )
+    lateinit var path: String
+
+    @Option(
+        names = ["-n", "--namespace"],
+        description = ["The namespace in which to look for both the Postgres and Kafka secrets"]
+    )
+    var namespace: String? = null
+
+    @Option(
+        names = ["-u", "--url"],
+        description = ["The kubernetes cluster URL (if the preinstall is being called from outside the cluster)"]
+    )
+    var url: String? = null
+
+    @Option(
+        names = ["-f", "--file"],
+        description = ["The file location of the truststore for Kafka"]
+    )
+    var truststoreLocation: String? = null
+
+    @Option(
+        names = ["-t", "--timeout"],
+        description = ["The timeout in milliseconds for testing the kafka connection - defaults to 3000"]
+    )
+    var timeout: Int = 3000
+
+    @Option(
+        names = ["-m", "--max-idle"],
+        description = ["The maximum ms a connection can be idle for while testing the kafka connection - defaults to 5000"]
```

Comment:
Is this something that we really need to expose?
```suggestion
        description = ["The maximum milliseconds a connection can be idle for while testing the Kafka connection - defaults to 5000"]
```